phpdotnet/phd/OutputHandler.php

```diff
@@ -0,0 +1,117 @@
+<?php
+namespace phpdotnet\phd;
+
+class OutputHandler
+{
+    public function __construct(
+        private Config $config
+    ) {}
+
+    /**
+     * Method to get a color escape sequence
+     */
+    private function term_color(string $text, string|bool $color): string {
+        return $this->config->color_output() && $color !== false ? "\033[" . $color . "m" . $text . "\033[m" : $text;
+    }
+
+    public function printPhdInfo(string $msg, ?string $info = ""): int {
+        $color = $this->config->phd_info_color();
+        $outputStream = $this->config->phd_info_output();
+
+        return $this->print($msg, $outputStream, $color, $info);
+    }
+
+    private function printPhdWarning(string $msg, ?string $warning = ""): int {
+        $color = $this->config->phd_warning_color();
+        $outputStream = $this->config->phd_warning_output();
+
+        return $this->print($msg, $outputStream, $color, $warning);
+    }
+
+    public function printUserError(string $msg, string $file, int $line, ?string $error = ""): int {
+        $color = $this->config->user_error_color();
+        $outputStream = $this->config->user_error_output();
+        $data = \sprintf("%s:%d\n\t%s", $file, $line, $msg);
+
+        return $this->print($data, $outputStream, $color, $error);
+    }
+
+    public function printPhpError(string $msg, string $file, int $line, ?string $error = ""): int {
+        $color = $this->config->php_error_color();
+        $outputStream = $this->config->php_error_output();
+        $data = \sprintf("%s:%d\n\t%s", $file, $line, $msg);
+
+        return $this->print($data, $outputStream, $color, $error);
+    }
+
+    private function print(string $msg, $outputStream, string|false $color = false, ?string $infoOrErrorString = ""): int {
```

Comment:
Ditto for `$infoOrErrorString`, especially as it doesn't seem like you are handling the `null` case at all.